docker-build.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+# define paths
+SCRIPT=$(readlink -f "$0")
+DIR=$(dirname "$SCRIPT")
+BUILD_COMMON="$DIR/common"
+BUILD_FBG="$DIR/fbg-server"
+EXPORT_FILE="freeboardgames.tar"
+
+# ensure yarn and npm are installed
+yarn -v > /dev/null 2>&1 || (echo "ERROR: yarn is not installed." && exit 1)
+npm -v > /dev/null 2>&1  || (echo "ERROR: npm is not installed." && exit 1)
+
+# define functions
+confirm() {
+    # call with a prompt string or use a default
+    read -r -p "$@"" [y/N]: " response
+    case "$response" in
+        [yY][eE][sS]|[yY])
+            return 0
+            ;;
+        *)
+            return 1
+            ;;
+    esac
+}
+
+install_fbg() {
+    cd "$BUILD_FBG"
+    read -p "Now fbg will be installed and started. Shut down the server after startup with CTRL+C. [PROCEED]"
+    yarn install && yarn start || exit 1
+}
+
+run_gameserver() {
+    cd "$DIR"
+    read -p "Now the webserver and backend will start. Shut down the server after startup with CTRL+C. [PROCEED]"
+    yarn run dev || (echo -e "ERROR. (ensure node is up-to-date)" && exit 1)
+}
+
+install_dependencies() {
+    cd "$DIR"
+    yarn install || (echo -e "ERROR. (ensure node is up-to-date)" && exit 1)
+}
+
+compile_dependencies() {
+    cd "$DIR"
+    install_dependencies && run_gameserver || exit 1
+}
+
+build_common_docker() {
```

Comment:
This is great, thank you :). Do you mind deleting the existing scripts on the common and web folder since this doesn't need it?

Reply:
Done - did cleanup a bit.